types/wlr_idle.c

```diff
@@ -101,34 +100,44 @@ static void handle_input_notification(struct wl_listener *listener, void *data)
 	}
 }
 
-static void create_idle_timer(struct wl_client *client,
-		struct wl_resource *idle_resource, uint32_t id,
-		struct wl_resource *seat_resource, uint32_t timeout) {
-	struct wlr_idle *idle = idle_from_resource(idle_resource);
-	struct wlr_seat_client *client_seat =
-		wlr_seat_client_from_resource(seat_resource);
-
+/* Creates a new timer. If idle_resource is not NULL, then id is used to create
+ * a new idle resource for the timer */
+static struct wlr_idle_timeout *create_timer(struct wlr_idle *idle,
+		struct wlr_seat *seat, uint32_t timeout,
+		struct wl_resource *idle_resource, uint32_t id) {
 	struct wlr_idle_timeout *timer =
 		calloc(1, sizeof(struct wlr_idle_timeout));
 	if (!timer) {
-		wl_resource_post_no_memory(idle_resource);
-		return;
+		if (idle_resource) {
+			wl_resource_post_no_memory(idle_resource);
+		}
+		return NULL;
+	}
+
+	if (idle_resource)
+	{
+		timer->resource = wl_resource_create(
```

Comment:
Hmm. Can we just create the resource in the caller, and `wl_resource_set_user_data(resource, timer)` here?

Reply:
You're right, makes more sense. I totally forgot I can just check user data later.